perception/autoware_tensorrt_bevdet/src/postprocess.cu

```diff
@@ -0,0 +1,212 @@
+    float * dim = (float *)bev_buffer[i * 6 + 2];      // 3 x 128 x 128
+    float * rot = (float *)bev_buffer[i * 6 + 3];      // 2 x 128 x 128
+    float * vel = (float *)bev_buffer[i * 6 + 4];      // 2 x 128 x 128
+    float * heatmap = (float *)bev_buffer[i * 6 + 5];  // c x 128 x 128
+
+    dim3 grid(DIVUP(map_size, NUM_THREADS));
+    CHECK_CUDA(cudaMemset(valid_box_num, 0, sizeof(int)));
+    BEVDecodeObjectKernel<<<grid, NUM_THREADS>>>(
+      map_size, score_thresh, x_start, y_start, x_step, y_step, output_h, output_w, down_sample,
+      class_num_pre_task[i], cur_start_label, reg, hei, dim, rot, vel, heatmap, boxes_dev,
+      score_dev, cls_dev, valid_box_num, nms_rescale_factor_dev);
+
+    /*
+    at this point, boxes_dev, score_dev, cls_dev have valid_box_num elements，which may be greater
+    than nms_pre_maxnum, and it's arranged in disorder
+    */
+    int box_num_pre = 0;
+    CHECK_CUDA(cudaMemcpy(&box_num_pre, valid_box_num, sizeof(int), cudaMemcpyDeviceToHost));
+
+    thrust::sequence(thrust::device, sorted_indices_dev, sorted_indices_dev + box_num_pre);
+    thrust::sort_by_key(
+      thrust::device, score_dev, score_dev + box_num_pre, sorted_indices_dev,
+      thrust::greater<float>());
+    // score_dev is sorted in descending order, while sorted_indices_dev indexes the original
+    // order,， sorted_indices_dev[i] = j; i:current，j:pre-index;  j:[0, map_size]
+
+    box_num_pre = std::min(box_num_pre, nms_pre_maxnum);
+
+    int box_num_post =
+      iou3d_nms->DoIou3dNms(box_num_pre, boxes_dev, sorted_indices_dev, keep_data_host);
+
+    box_num_post = std::min(box_num_post, nms_post_maxnum);
+
+    CHECK_CUDA(cudaMemcpy(
+      sorted_indices_host, sorted_indices_dev, box_num_pre * sizeof(int), cudaMemcpyDeviceToHost));
+    CHECK_CUDA(cudaMemcpy(
+      boxes_host, boxes_dev, kBoxBlockSize * map_size * sizeof(float), cudaMemcpyDeviceToHost));
+    CHECK_CUDA(
+      cudaMemcpy(score_host, score_dev, box_num_pre * sizeof(float), cudaMemcpyDeviceToHost));
+    CHECK_CUDA(cudaMemcpy(cls_host, cls_dev, map_size * sizeof(float), cudaMemcpyDeviceToHost));
+
+    for (auto j = 0; j < box_num_post; j++) {
+      int k = keep_data_host[j];
+      int idx = sorted_indices_host[k];
+      Box box;
+      box.x = boxes_host[idx * kBoxBlockSize + 0];
+      box.y = boxes_host[idx * kBoxBlockSize + 1];
+      box.z = boxes_host[idx * kBoxBlockSize + 2];
+      box.l = boxes_host[idx * kBoxBlockSize + 3] / nms_rescale_factor[cls_host[idx]];
+      box.w = boxes_host[idx * kBoxBlockSize + 4] / nms_rescale_factor[cls_host[idx]];
+      box.h = boxes_host[idx * kBoxBlockSize + 5] / nms_rescale_factor[cls_host[idx]];
+      box.r = boxes_host[idx * kBoxBlockSize + 6];
+      box.vx = boxes_host[idx * kBoxBlockSize + 7];
+      box.vy = boxes_host[idx * kBoxBlockSize + 8];
+
+      box.label = cls_host[idx];
+      box.score = score_host[k];
+      box.z -= box.h * 0.5;  // bottom height
```

Comment:
Consider adding a comment explaining why `box.z` is adjusted by `box.h * 0.5`. This will improve code readability.